Versioning Proposal

AR21SM opened the pull request:
Added My proposal for Versioning System for DISCOVER Cookbook

AR21SM commented:
Hi @aterrel, @kamila-NF, @melissawm,

I have added my proposal for the Versioning System in the DISCOVER Cookbook. 

Looking forward to your feedback. 

Best regards,
AR21SM

AR21SM commented:
Hi @aterrel @melissawm,

I wanted to follow up on my GSoC proposal for the Versioning System for DISCOVER Cookbook that I submitted last week (21-March ). I understand you're busy with your regular work commitments, and I appreciate your time.

I'm eager to receive any feedback you might have on my submission. If there are any aspects of the proposal that need clarification or additional information, I would be happy to address them.

Thank you for your time and consideration.

Have a nice day, 
AR21SM

AR21SM commented:
Hi @aterrel,

I wanted to show my progress if u have  2-3 minutes can u see it :

1. I've created a version banner for older editions of the Cookbook and shared a demo video in discussion #236
2. I've also implemented a toggle button feature with a demo video shared in Discussion #219

When you have a moment, I'd appreciate your feedback on these implementations. Thanks for your time!

(**Just wanted to let you know that I've submitted my proposal for the DISCOVER Cookbook project on the Google Summer of Code website.**)

-AR21SM

aterrel commented:
@AR21SM what is the name of your proposal on gsoc?

aterrel commented on docs/proposals/Ashish_Mahajan_Versioning_Proposal.md:
```diff
@@ -0,0 +1,241 @@
+
+The DISCOVER Cookbook currently maintains two separate versions with no navigation between them:
+
+- Current version (v2.0) in the main branch, deployed at https://discover-cookbook.numfocus.org
```
No there is just the main branch. The master branch is just an old stale branch from when git changed the default branch name.

aterrel commented on docs/proposals/Ashish_Mahajan_Versioning_Proposal.md:
```diff
@@ -0,0 +1,241 @@
+
+My implementation follows the industry-standard branch-based versioning model used by scikit-learn, Matplotlib, NetworkX, and other NumFOCUS projects:
+
+1. **Branch-Version Mapping**:
```
It's not the best design to make a branch for each of these. I prefer tags that will have tooling around the tag. 

For example you have pandas with only one active branch: https://github.com/pandas-dev/pandas/branches

but the repository has tools that allow it to present other branches: https://pandas.pydata.org/docs/

aterrel commented on docs/proposals/Ashish_Mahajan_Versioning_Proposal.md:
```diff
@@ -0,0 +1,241 @@
+
+### 1. Version Data Architecture
+
+I will implement a centralized version registry that defines all versions and their relationships. This architecture will:
```
no this would be too complex to manage. We really want to just have tags for the "editions" like a version of a software library.

aterrel commented on docs/proposals/Ashish_Mahajan_Versioning_Proposal.md:
```diff
@@ -0,0 +1,241 @@
+   - Version-specific metadata (status)
+   - Visual distinction for latest version
+
+4. **Responsive Design**:
```
seems like a stretch goal. I would rather not change the over arching design of the entire project.

aterrel commented on docs/proposals/Ashish_Mahajan_Versioning_Proposal.md:
```diff
@@ -0,0 +1,241 @@
+
+### 7. Implementation Dependencies
+
+- **Step 1** (Version Architecture) is foundational and must be implemented first
```
We need Step 0 which is a signed off design. As commented above the version architecture should just be tags.

aterrel reviewed (changes requested):
A few inline comments to be sorted.

AR21SM commented:
Thank you for taking the time to review my proposal and provide this valuable feedback.

I would like to request 3-4 days to study your suggestions thoroughly and revise my approach. I'm currently managing some college exams (quizs and lab evaluations), but I'm fully committed to addressing all your feedback points.

I'll share my progress with you soon, that addresses all your comments.

Thanks again for your guidance!
AR21SM

aterrel commented:
@AR21SM have you been able to address the requested changes?

AR21SM commented:
Hi @aterrel , apologies for the delay — I've now addressed all your feedback and updated the proposal, kindly requesting a re-review.

aterrel commented on docs/proposals/002-Versioning_Proposal.md:
```diff
@@ -0,0 +1,209 @@
+
+1. **Tag-Based Version Management:**:
+   - Maintain a single active branch (main)
+   - Use Git tags to mark specific versions (latest version , previous versions)
```
we should stick to numeric versions. 1.0, 2.0, and then the latest version will just be the main branch and can be labelled 2.0-dev. One might even look at how to included [versioneer](https://github.com/python-versioneer/python-versioneer) so you can see which version of the book is being renderd.

aterrel commented on docs/proposals/002-Versioning_Proposal.md:
```diff
@@ -0,0 +1,209 @@
+1. **Tag-Based Version Management:**:
+   - Maintain a single active branch (main)
+   - Use Git tags to mark specific versions (latest version , previous versions)
+   - Deploy each tagged version to its corresponding URL path
```
Here is where you will have a development url as well. Would probably want to have a separate script for local and deployment so that folks can still iterated on the content locally without much trouble.

aterrel commented on docs/proposals/002-Versioning_Proposal.md:
```diff
@@ -0,0 +1,209 @@
+   - Deploy each tagged version to its corresponding URL path
+
+2. **URL Structure**:
+   - Latest version: discover-cookbook.numfocus.org/
```
would be good to have a redirect for things like `discover-cookbook.numfocus.org/05_venue_selection/` to ``discover-cookbook.numfocus.org/v1.0/05_venue_selection/`

aterrel commented on docs/proposals/002-Versioning_Proposal.md:
```diff
@@ -0,0 +1,209 @@
+- Built-in versioning system
+- Established version selector UI
+
+**Disadvantages**:
```
A few others to add:

- This also requires a paid account to run ad free
- depends on a more complex service than just hosting static webpages, which can be done on a multitude of services.
